bindata/network/openshift-sdn/sdn.yaml

```diff
@@ -67,26 +67,45 @@ spec:
           #!/bin/bash
           set -euo pipefail
 {{- if .IsNetworkTypeLiveMigration }}
-          # In case of rollback, openshift-sdn need to reuse the host subnet used by ovn-kube, so we need to create the hostsubnet CR for the local node if it doesn't exist.
+          set -x
+          NODE_CNI="OpenShiftSDN"
+          if ip link show br-ex; then
+            echo "br-ex exists"
+            NODE_CNI="OVNKubernetes"
+          else
+            echo "br-ex doesn't exist"
+          fi
+
           # retry loop
           RETRY_INTERVAL=3
-          MAX_RETRIES=10
+          MAX_RETRIES=100
           for ((retry_count=1; retry_count<=MAX_RETRIES; retry_count++)); do
-              # Get SDN subnet
-              SDN_SUBNET=$(oc get hostsubnet ${K8S_NODE_NAME} -o jsonpath="{.subnet}"||true)
+            # Get SDN subnet
+            SDN_SUBNET=$(oc get hostsubnet ${K8S_NODE_NAME} -o jsonpath="{.subnet}"||true)
 
-              # Get OVN subnet
-              OVN_SUBNET=$(oc get node ${K8S_NODE_NAME} -o jsonpath='{.metadata.annotations.k8s\.ovn\.org/node-subnets}' | grep -o -E '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/[0-9]{1,3}'||true)
+            # Get OVN subnet
+            OVN_SUBNET=$(oc get node ${K8S_NODE_NAME} -o jsonpath='{.metadata.annotations.k8s\.ovn\.org/node-subnets}' | grep -o -E '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/[0-9]{1,3}'||true)
 
-              if [ -n "${SDN_SUBNET}" ] && [ -z "${OVN_SUBNET}" ]; then
-                  echo "SDN_SUBNET is ${SDN_SUBNET}"
-                  break
-              elif [ -z "${SDN_SUBNET}" ] && [ -n "${OVN_SUBNET}" ]; then
+            if [ "${NODE_CNI}" == "OpenShiftSDN" ]; then
+              if [ -z "${SDN_SUBNET}" ]; then
+                echo "SDN node subnet is not found, retry..."
+                sleep $RETRY_INTERVAL
+              else
+                break
+              fi
+            else
+              if [ -z "${OVN_SUBNET}" ]; then
+                echo "OVN node subnet is not found, retry..."
+                sleep $RETRY_INTERVAL
+              else
+                # When rolling back, SDN controll plane will not start until the hostsubnet has been created here.
+                echo "Use OVN-K subnet ${OVN_SUBNET}, when the node CNI is OVN-K"
+                if [ -z "${SDN_SUBNET}" ]; then
```

Comment:
Why are you no longer verifying that SDN_SUBNET is equal to OVN_SUBNET if both are set?
Wouldn't that cause issues when  a node is added during the migration to SDN and it is still using ovn-k but the sdn-controller allocates a subnet for it?

Reply:
In that case, ovnkube-node will use the node subnet allocated by SDN. So we don't care whether the SDN_SUBNET is equal to the OVN_SUBNET. It will be overwritten regardless. And here, the SDN pod will wait until the OVN_SUBNET is set by the ovnkube-node pod.

Reply:
This is about OVN-K -> SDN migration.
- A new node is added during the migration and it still uses ovn-k
- Both CNIs allocate the subnet
- In sdn pod `if [ -z "${SDN_SUBNET}" ]; then` is not going to be true and SDN will use the subnet allocated by SDN which can conflict with a subnet allocate by ovn-k.

Does this make sense? In the current code we are not going to overwrite the hostsubnet if there is already one specified.

Reply:
We synced offline, here is how it would look like for sdn->ovn-k migration:
- Migration starts and both controllers started after syncing on current subnets (sdn container created all hostsubnets from ovn subnets)
- A new node is added, both controllers allocate a subnet (it shouldn't conflict with any existign subnets but can be different for each CNI)
- With the changes from this PR the ovn-k pod will first overwrite the ovn-k subnet with the value allocated by SDN